### ISSUE TITLE
Layers and overlaying widgets (apparent change from 0.14 to 0.15)

### ISSUE THREAD

**Issue by davep** [Textualize/textual#2081]
Turning the following discussion into an issue to be looked at as a possible bug. The following code works as the author expected in 0.14, but seems to not work well at all in 0.15. Creating an issue as I feel this deserves testing from a potential bug point of view.

### Discussed in https://github.com/Textualize/textual/discussions/2076

<div type='discussions-op-text'>

<sup>Originally posted by **MaineTim** March 15, 2023</sup>
I'm taking a poke at playing with layers, with the goal of being able to overlay an input widget with an a second input widget, and then remove the second revealing the first still intact. In a modified version of the layers example:

```

from textual.app import App, ComposeResult
from textual.widgets import Static


class LayersExample(App):
    CSS_PATH = "layers.css"

    BINDINGS = [
        ("t", "toggle", "")
    ]

    def compose(self) -> ComposeResult:
        yield Static("box2 (layer = below)", id="box2")

    def on_mount(self):
        self.widget_show = False

    def action_toggle(self):
        self.log("Toggled!")
        self.widget_show = False if self.widget_show else True
        if self.widget_show:
            self.mount(Static("box1 (layer = above)", id="box1"))
        else:
            self.query_one("#box1").remove()
            self.set_focus(self.query_one("#box2"))



if __name__ == "__main__":
    app = LayersExample()
    app.run()

```
and layers.css:
```
Screen {
    align: center middle;
    layers: below above;
}

Static {
    width: 28;
    height: 8;
    color: auto;
    content-align: center middle;
}

#box1 {
    layer: above;
    background: darkcyan;
}

#box2 {
    layer: below;
    background: orange;
}

```
When I run the program and hit T, box1 overlays box2 correctly. When I toggle again, box1 is removed from the DOM, but is still present on the screen. If I sweep the mouse back and forth across box1, it will disappear and box2 reappears. I guess I'm triggering a screen update by swiping the mouse across the now removed widget?

I also attempted this same idea using the the visibility attribute of the higher-layer widget rather than removing it completely, and it suffered from the same lack of an update.

How can I cause that update programmatically? Or am I just going about this the wrong way?

Thanks for any advice!
</div>

**Comment by MaineTim** [comment]
To follow up on this with another symptom: I just noticed that when popping a screen, the footer is not updated (returned back to the previous screen footer) until I sweep the mouse across it.

**Comment by TomJGooding** [comment]
It looks like this was fixed back in v0.16 so this issue can probably now be closed. Spotted when looking back through issues for something else.

**Comment by davep** [comment]
Closing as complete (thanks for checking way back when Tom!)
